lib/jxl/dec_modular.cc

```diff
@@ -589,19 +590,33 @@ Status ModularFrameDecoder::ModularImageToDecodedRect(
     const size_t ec_ysize = DivCeil(frame_dim.ysize_upsampled, ecups);
     JXL_ASSERT(c < gi.channel.size());
     Channel& ch_in = gi.channel[c];
-    // For x0, y0 there's no need to do a DivCeil().
-    JXL_DASSERT(rect.x0() % (1ul << ch_in.hshift) == 0);
-    JXL_DASSERT(rect.y0() % (1ul << ch_in.vshift) == 0);
+    if (!render_pipeline_input) {
```

Comment:
Why make the DASSERTs conditional?

Reply:
Because `rect` is not used and may be invalid in the render_pipeline_input case :)